### ISSUE TITLE
forwardRotationLambda does not normalize angles beyond ±3π

### ISSUE THREAD

**Issue by jrus** [d3/d3-geo#242]
If you try to project angles outside the range [–3π, 3π], it can result in wonky artifacts from resampling:

<img width="934" alt="Screen Shot 2021-06-20 at 9 05 45 PM" src="https://user-images.githubusercontent.com/83411/122713955-624b8b00-d21b-11eb-9d36-f0510e7004ae.png">

Here is the current code for `forwardRotationLambda` https://github.com/d3/d3-geo/blob/v3.0.1/src/rotation.js#L17-L21

```js
function forwardRotationLambda(deltaLambda) {
  return function(lambda, phi) {
    return lambda += deltaLambda, [lambda > pi ? lambda - tau : lambda < -pi ? lambda + tau : lambda, phi];
  };
}
```

My proposed alternative is:

```js
function forwardRotationLambda(deltaLambda) {
  return function(lambda, phi) {
    lambda += deltaLambda;
    while (lambda > pi) lambda -= tau;
    while (lambda < -pi) lambda += tau;
    return [lambda, phi];
  };
}
```

This should run at the same speed, and in my opinion is also a bit more legible.

**Comment by jrus** [comment]
I suppose one potential issue with a loop would be dealing with cases where someone puts the angle as `1e30` or something.

**Comment by Fil** [comment]
Hmmm… maybe it's the Mercator projection that's taking too short a shortcut? I wasn't able to reproduce though with d3.geoMercator.

**Comment by jrus** [comment]
Example:

```js
{
  const width = 600;
  const segment = ({ type: "LineString",
    coordinates: [[-800, 55], [-1000, -35]] })
  const sphere = ({type: "Sphere"});
  const projection = d3.geoMercator().rotate([52]).precision(0.01)
    .fitExtent([[5,5], [width-5, width-5]], sphere)
  
  const context = DOM.context2d(width, width);
  const path = d3.geoPath(projection, context);
  
  context.beginPath(), path(d3.geoGraticule10()),
    context.lineWidth = 1.5; context.strokeStyle = `#ccc`, context.stroke();
  context.beginPath(), path(segment),
    context.lineWidth = 6; context.strokeStyle = "#26b", context.stroke();
  context.beginPath(), path(sphere),
    context.lineWidth = 2; context.strokeStyle = "#000", context.stroke();
  return context.canvas;

}
```

**Comment by jrus** [comment]
Oh, I guess the identity rotation removes the appropriate number of turns:  https://github.com/d3/d3-geo/blob/v3.0.1/src/rotation.js#L4-L6

```js
function rotationIdentity(lambda, phi) {
  return [abs(lambda) > pi ? lambda + Math.round(-lambda / tau) * tau : lambda, phi];
}
```

So we could also just use that method for `forwardRotationLambda`.

**Comment by Fil** [comment]
I was wondering if the alternative was to fix mercatorRaw instead (in https://github.com/d3/d3-geo/blob/v3.0.1/src/projection/mercator.js#L6), but the same issue impacts all the projections that use lambda without trig, for example geoNaturalEarth1 and geoEquirectangular.

I guess https://github.com/d3/d3-geo/issues/147 should have covered forwardRotationLambda too!

**Comment by jrus** [comment]
Is there a preferred style between e.g.

```js
function forwardRotationLambda(deltaLambda) {
  return function(lambda, phi) {
    lambda += deltaLambda;
    if (abs(lambda) > pi) lambda -= Math.round(lambda / tau) * tau;
    return [lambda, phi];
  };
}
```

```js
function forwardRotationLambda(deltaLambda) {
  return function(lambda, phi) {
    return [abs(lambda += deltaLambda) > pi ? lambda - Math.round(lambda / tau) * tau : lambda, phi];
  };
}
```

```js
function forwardRotationLambda(deltaLambda) {
  return function(lambda, phi) {
    return [(lambda += deltaLambda) - (abs(lambda) > pi ? Math.round(lambda / tau) * tau : 0), phi];
  };
}
```

I find the crammed-in-one-line versions less legible, but they seem closer to the prevailing style.

**Comment by Fil** [comment]
I'm happy with the first version.

**Comment by mbostock** [comment]
Fixed in #244.
